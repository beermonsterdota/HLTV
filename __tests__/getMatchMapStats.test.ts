import HLTV from '../src/index.js'
import { sleep } from '../src/utils.js'
import { expect, test } from 'vitest'

const FULL_INFO_NEWEST = 116688
const FULL_INFO = 79440
const RATING_1 = 14783

test('getMatchMapStats', async () => {
  await sleep(10000)
  expect(
    await HLTV.getMatchMapStats({ id: FULL_INFO_NEWEST })
  ).toMatchSnapshot()
  await sleep(3000)
  expect(await HLTV.getMatchMapStats({ id: FULL_INFO })).toMatchSnapshot()
  await sleep(3000)
  expect(await HLTV.getMatchMapStats({ id: RATING_1 })).toMatchSnapshot()
  await sleep(3000)
}, 30000)
