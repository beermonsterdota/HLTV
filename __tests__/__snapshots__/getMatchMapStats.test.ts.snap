// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getMatchMapStats 1`] = `
{
  "date": 1616065500000,
  "event": {
    "id": 5785,
    "name": "Snow Sweet Snow 3 Regionals",
  },
  "id": 116688,
  "map": "de_vertigo",
  "matchId": 2347210,
  "overview": {
    "bestRating2": {
      "id": 16410,
      "name": "Duplicate",
      "value": 1.53,
    },
    "clutchesWon": {
      "team1": 1,
      "team2": 1,
    },
    "firstKills": {
      "team1": 18,
      "team2": 8,
    },
    "mostAWPKills": {
      "id": 12379,
      "name": "Impulse",
      "value": 19,
    },
    "mostAssists": {
      "id": 3730,
      "name": "HOLMES",
      "value": 7,
    },
    "mostDamage": {
      "id": 12379,
      "name": "Impulse",
      "value": 108,
    },
    "mostFirstKills": {
      "id": 13466,
      "name": "dennyslaw",
      "value": 8,
    },
    "mostKills": {
      "id": 12379,
      "name": "Impulse",
      "value": 30,
    },
    "rating": {
      "team1": 1.27,
      "team2": 0.85,
    },
  },
  "performanceOverview": null,
  "playerStats": {
    "team1": [
      {
        "ADR": 107.9,
        "KAST": 76.9,
        "assists": 8,
        "deaths": 15,
        "firstKillsDifference": -3,
        "flashAssists": 1,
        "hsKills": 11,
        "killDeathsDifference": 8,
        "kills": 23,
        "player": {
          "id": 16410,
          "name": "Duplicate",
        },
        "rating1": 1.53,
      },
      {
        "ADR": 71,
        "KAST": 80.8,
        "assists": 4,
        "deaths": 15,
        "firstKillsDifference": 6,
        "flashAssists": 3,
        "hsKills": 13,
        "killDeathsDifference": 8,
        "kills": 23,
        "player": {
          "id": 13466,
          "name": "dennyslaw",
        },
        "rating1": 1.4,
      },
      {
        "ADR": 73.5,
        "KAST": 80.8,
        "assists": 6,
        "deaths": 13,
        "firstKillsDifference": 1,
        "flashAssists": 1,
        "hsKills": 7,
        "killDeathsDifference": 6,
        "kills": 19,
        "player": {
          "id": 17145,
          "name": "Rainwaker",
        },
        "rating1": 1.21,
      },
      {
        "ADR": 65,
        "KAST": 80.8,
        "assists": 1,
        "deaths": 15,
        "firstKillsDifference": 3,
        "flashAssists": 0,
        "hsKills": 10,
        "killDeathsDifference": 2,
        "kills": 17,
        "player": {
          "id": 18697,
          "name": "oxygeN",
        },
        "rating1": 1.09,
      },
      {
        "ADR": 75.8,
        "KAST": 73.1,
        "assists": 7,
        "deaths": 18,
        "firstKillsDifference": 3,
        "flashAssists": 2,
        "hsKills": 9,
        "killDeathsDifference": -2,
        "kills": 16,
        "player": {
          "id": 11343,
          "name": "SHiPZ",
        },
        "rating1": 1.09,
      },
    ],
    "team2": [
      {
        "ADR": 108,
        "KAST": 69.2,
        "assists": 5,
        "deaths": 17,
        "firstKillsDifference": -1,
        "flashAssists": 4,
        "hsKills": 7,
        "killDeathsDifference": 13,
        "kills": 30,
        "player": {
          "id": 12379,
          "name": "Impulse",
        },
        "rating1": 1.5,
      },
      {
        "ADR": 53.2,
        "KAST": 61.5,
        "assists": 2,
        "deaths": 20,
        "firstKillsDifference": -3,
        "flashAssists": 0,
        "hsKills": 9,
        "killDeathsDifference": -5,
        "kills": 15,
        "player": {
          "id": 3732,
          "name": "DiMKE",
        },
        "rating1": 0.76,
      },
      {
        "ADR": 67.1,
        "KAST": 53.8,
        "assists": 3,
        "deaths": 21,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 6,
        "killDeathsDifference": -8,
        "kills": 13,
        "player": {
          "id": 5351,
          "name": "aVN",
        },
        "rating1": 0.71,
      },
      {
        "ADR": 51.5,
        "KAST": 61.5,
        "assists": 5,
        "deaths": 19,
        "firstKillsDifference": -1,
        "flashAssists": 4,
        "hsKills": 7,
        "killDeathsDifference": -8,
        "kills": 11,
        "player": {
          "id": 4218,
          "name": "Dav",
        },
        "rating1": 0.68,
      },
      {
        "ADR": 53.5,
        "KAST": 61.5,
        "assists": 9,
        "deaths": 21,
        "firstKillsDifference": -4,
        "flashAssists": 2,
        "hsKills": 5,
        "killDeathsDifference": -14,
        "kills": 7,
        "player": {
          "id": 3730,
          "name": "HOLMES",
        },
        "rating1": 0.58,
      },
    ],
  },
  "result": {
    "halfResults": [
      {
        "team1Rounds": 11,
        "team2Rounds": 4,
      },
      {
        "team1Rounds": 5,
        "team2Rounds": 6,
      },
    ],
    "team1TotalRounds": 16,
    "team2TotalRounds": 10,
  },
  "roundHistory": [
    {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "1-0",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "2-0",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "3-0",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "bomb_exploded",
      "score": "4-0",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "bomb_exploded",
      "score": "5-0",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "6-0",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "7-0",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "bomb_exploded",
      "score": "8-0",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "9-0",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "ct_win",
      "score": "9-1",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "ct_win",
      "score": "9-2",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "10-2",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "bomb_defused",
      "score": "10-3",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "ct_win",
      "score": "10-4",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "11-4",
      "tTeam": 10386,
    },
    {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "11-5",
      "tTeam": 10386,
    },
    {
      "ctTeam": 10386,
      "outcome": "t_win",
      "score": "11-6",
      "tTeam": 11001,
    },
    {
      "ctTeam": 10386,
      "outcome": "t_win",
      "score": "11-7",
      "tTeam": 11001,
    },
    {
      "ctTeam": 10386,
      "outcome": "t_win",
      "score": "11-8",
      "tTeam": 11001,
    },
    {
      "ctTeam": 10386,
      "outcome": "ct_win",
      "score": "12-8",
      "tTeam": 11001,
    },
    {
      "ctTeam": 10386,
      "outcome": "stopwatch",
      "score": "13-8",
      "tTeam": 11001,
    },
    {
      "ctTeam": 10386,
      "outcome": "bomb_defused",
      "score": "14-8",
      "tTeam": 11001,
    },
    {
      "ctTeam": 10386,
      "outcome": "bomb_exploded",
      "score": "14-9",
      "tTeam": 11001,
    },
    {
      "ctTeam": 10386,
      "outcome": "t_win",
      "score": "14-10",
      "tTeam": 11001,
    },
    {
      "ctTeam": 10386,
      "outcome": "ct_win",
      "score": "15-10",
      "tTeam": 11001,
    },
    {
      "ctTeam": 10386,
      "outcome": "ct_win",
      "score": "16-10",
      "tTeam": 11001,
    },
  ],
  "team1": {
    "id": 10386,
    "name": "SKADE",
  },
  "team2": {
    "id": 11001,
    "name": "ONYX",
  },
}
`;

exports[`getMatchMapStats 2`] = `
{
  "date": 1544901600000,
  "event": {
    "id": 4205,
    "name": "BLAST Pro Series Lisbon 2018",
  },
  "id": 79440,
  "map": "de_overpass",
  "matchId": 2329728,
  "overview": {
    "bestRating2": {
      "id": 7998,
      "name": "s1mple",
      "value": 1.61,
    },
    "clutchesWon": {
      "team1": 0,
      "team2": 3,
    },
    "firstKills": {
      "team1": 10,
      "team2": 13,
    },
    "mostAWPKills": {
      "id": 7998,
      "name": "s1mple",
      "value": 9,
    },
    "mostAssists": {
      "id": 484,
      "name": "Zeus",
      "value": 9,
    },
    "mostDamage": {
      "id": 7398,
      "name": "dupreeh",
      "value": 99.9,
    },
    "mostFirstKills": {
      "id": 8918,
      "name": "electroNic",
      "value": 5,
    },
    "mostKills": {
      "id": 7398,
      "name": "dupreeh",
      "value": 23,
    },
    "rating": {
      "team1": 0.81,
      "team2": 1.28,
    },
  },
  "performanceOverview": null,
  "playerStats": {
    "team1": [
      {
        "ADR": 99.9,
        "KAST": 78.3,
        "assists": 1,
        "deaths": 19,
        "firstKillsDifference": 1,
        "flashAssists": 0,
        "hsKills": 9,
        "killDeathsDifference": 4,
        "kills": 23,
        "player": {
          "id": 7398,
          "name": "dupreeh",
        },
        "rating1": 1.32,
      },
      {
        "ADR": 78.7,
        "KAST": 56.5,
        "assists": 4,
        "deaths": 17,
        "firstKillsDifference": 0,
        "flashAssists": 1,
        "hsKills": 9,
        "killDeathsDifference": -3,
        "kills": 14,
        "player": {
          "id": 9032,
          "name": "Magisk",
        },
        "rating1": 0.89,
      },
      {
        "ADR": 70,
        "KAST": 60.9,
        "assists": 3,
        "deaths": 19,
        "firstKillsDifference": 0,
        "flashAssists": 1,
        "hsKills": 4,
        "killDeathsDifference": -7,
        "kills": 12,
        "player": {
          "id": 7592,
          "name": "device",
        },
        "rating1": 0.72,
      },
      {
        "ADR": 50.3,
        "KAST": 52.2,
        "assists": 3,
        "deaths": 16,
        "firstKillsDifference": -3,
        "flashAssists": 1,
        "hsKills": 4,
        "killDeathsDifference": -7,
        "kills": 9,
        "player": {
          "id": 7412,
          "name": "gla1ve",
        },
        "rating1": 0.56,
      },
      {
        "ADR": 54.2,
        "KAST": 52.2,
        "assists": 2,
        "deaths": 19,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 1,
        "killDeathsDifference": -12,
        "kills": 7,
        "player": {
          "id": 4954,
          "name": "Xyp9x",
        },
        "rating1": 0.53,
      },
    ],
    "team2": [
      {
        "ADR": 98.6,
        "KAST": 78.3,
        "assists": 1,
        "deaths": 11,
        "firstKillsDifference": 1,
        "flashAssists": 0,
        "hsKills": 12,
        "killDeathsDifference": 12,
        "kills": 23,
        "player": {
          "id": 7998,
          "name": "s1mple",
        },
        "rating1": 1.61,
      },
      {
        "ADR": 93,
        "KAST": 73.9,
        "assists": 4,
        "deaths": 14,
        "firstKillsDifference": 2,
        "flashAssists": 1,
        "hsKills": 11,
        "killDeathsDifference": 7,
        "kills": 21,
        "player": {
          "id": 8918,
          "name": "electroNic",
        },
        "rating1": 1.48,
      },
      {
        "ADR": 82.4,
        "KAST": 91.3,
        "assists": 2,
        "deaths": 12,
        "firstKillsDifference": 3,
        "flashAssists": 0,
        "hsKills": 8,
        "killDeathsDifference": 8,
        "kills": 20,
        "player": {
          "id": 483,
          "name": "Edward",
        },
        "rating1": 1.41,
      },
      {
        "ADR": 68.3,
        "KAST": 60.9,
        "assists": 4,
        "deaths": 13,
        "firstKillsDifference": -2,
        "flashAssists": 2,
        "hsKills": 8,
        "killDeathsDifference": 2,
        "kills": 15,
        "player": {
          "id": 7594,
          "name": "flamie",
        },
        "rating1": 1,
      },
      {
        "ADR": 74,
        "KAST": 69.6,
        "assists": 11,
        "deaths": 15,
        "firstKillsDifference": -1,
        "flashAssists": 2,
        "hsKills": 4,
        "killDeathsDifference": -4,
        "kills": 11,
        "player": {
          "id": 484,
          "name": "Zeus",
        },
        "rating1": 0.91,
      },
    ],
  },
  "result": {
    "halfResults": [
      {
        "team1Rounds": 4,
        "team2Rounds": 11,
      },
      {
        "team1Rounds": 3,
        "team2Rounds": 5,
      },
    ],
    "team1TotalRounds": 7,
    "team2TotalRounds": 16,
  },
  "roundHistory": [
    {
      "ctTeam": 4608,
      "outcome": "bomb_defused",
      "score": "1-0",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "t_win",
      "score": "1-1",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "t_win",
      "score": "1-2",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "t_win",
      "score": "1-3",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "bomb_exploded",
      "score": "1-4",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "t_win",
      "score": "1-5",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "stopwatch",
      "score": "2-5",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "t_win",
      "score": "2-6",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "t_win",
      "score": "2-7",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "t_win",
      "score": "2-8",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "ct_win",
      "score": "3-8",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "stopwatch",
      "score": "4-8",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "t_win",
      "score": "4-9",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "t_win",
      "score": "4-10",
      "tTeam": 6665,
    },
    {
      "ctTeam": 4608,
      "outcome": "t_win",
      "score": "4-11",
      "tTeam": 6665,
    },
    {
      "ctTeam": 6665,
      "outcome": "ct_win",
      "score": "4-12",
      "tTeam": 4608,
    },
    {
      "ctTeam": 6665,
      "outcome": "ct_win",
      "score": "4-13",
      "tTeam": 4608,
    },
    {
      "ctTeam": 6665,
      "outcome": "bomb_exploded",
      "score": "5-13",
      "tTeam": 4608,
    },
    {
      "ctTeam": 6665,
      "outcome": "bomb_exploded",
      "score": "6-13",
      "tTeam": 4608,
    },
    {
      "ctTeam": 6665,
      "outcome": "ct_win",
      "score": "6-14",
      "tTeam": 4608,
    },
    {
      "ctTeam": 6665,
      "outcome": "bomb_exploded",
      "score": "7-14",
      "tTeam": 4608,
    },
    {
      "ctTeam": 6665,
      "outcome": "bomb_defused",
      "score": "7-15",
      "tTeam": 4608,
    },
    {
      "ctTeam": 6665,
      "outcome": "ct_win",
      "score": "7-16",
      "tTeam": 4608,
    },
  ],
  "team1": {
    "id": 6665,
    "name": "Astralis",
  },
  "team2": {
    "id": 4608,
    "name": "Natus Vincere",
  },
}
`;

exports[`getMatchMapStats 3`] = `
{
  "date": 1370372400000,
  "event": {
    "id": 1147,
    "name": "RaidCall EMS One Summer 2013 Cup #4",
  },
  "id": 14783,
  "map": "de_inferno",
  "matchId": 2249891,
  "overview": {
    "bestRating1": {
      "id": 735,
      "name": "Dosia",
      "value": 1.3,
    },
    "clutchesWon": {
      "team1": 2,
      "team2": 2,
    },
    "firstKills": {
      "team1": 13,
      "team2": 16,
    },
    "mostAWPKills": {
      "id": 2757,
      "name": "GuardiaN",
      "value": 15,
    },
    "mostAssists": {
      "id": 7154,
      "name": "Pimp",
      "value": 7,
    },
    "mostFirstKills": {
      "id": 7413,
      "name": "Nico",
      "value": 6,
    },
    "mostKills": {
      "id": 735,
      "name": "Dosia",
      "value": 24,
    },
    "rating": {
      "team1": 0.91,
      "team2": 1.11,
    },
  },
  "performanceOverview": null,
  "playerStats": {
    "team1": [
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 7,
        "deaths": 19,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 7,
        "killDeathsDifference": -1,
        "kills": 18,
        "player": {
          "id": 7154,
          "name": "Pimp",
        },
        "rating1": 1.04,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 3,
        "deaths": 21,
        "firstKillsDifference": 1,
        "flashAssists": 0,
        "hsKills": 3,
        "killDeathsDifference": -1,
        "kills": 20,
        "player": {
          "id": 7413,
          "name": "Nico",
        },
        "rating1": 0.98,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 2,
        "deaths": 21,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 11,
        "killDeathsDifference": -1,
        "kills": 20,
        "player": {
          "id": 6796,
          "name": "Nille",
        },
        "rating1": 0.96,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 4,
        "deaths": 19,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 10,
        "killDeathsDifference": -3,
        "kills": 16,
        "player": {
          "id": 7156,
          "name": "MSL",
        },
        "rating1": 0.84,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 6,
        "deaths": 23,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 4,
        "killDeathsDifference": -7,
        "kills": 16,
        "player": {
          "id": 7412,
          "name": "gla1ve",
        },
        "rating1": 0.75,
      },
    ],
    "team2": [
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 3,
        "deaths": 16,
        "firstKillsDifference": 1,
        "flashAssists": 0,
        "hsKills": 16,
        "killDeathsDifference": 8,
        "kills": 24,
        "player": {
          "id": 735,
          "name": "Dosia",
        },
        "rating1": 1.3,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 4,
        "deaths": 18,
        "firstKillsDifference": 2,
        "flashAssists": 0,
        "hsKills": 1,
        "killDeathsDifference": 3,
        "kills": 21,
        "player": {
          "id": 2757,
          "name": "GuardiaN",
        },
        "rating1": 1.19,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 5,
        "deaths": 19,
        "firstKillsDifference": 3,
        "flashAssists": 0,
        "hsKills": 11,
        "killDeathsDifference": 4,
        "kills": 23,
        "player": {
          "id": 302,
          "name": "kUcheR",
        },
        "rating1": 1.16,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 1,
        "deaths": 17,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 12,
        "killDeathsDifference": 2,
        "kills": 19,
        "player": {
          "id": 737,
          "name": "Fox",
        },
        "rating1": 1.05,
      },
      {
        "ADR": undefined,
        "KAST": undefined,
        "assists": 6,
        "deaths": 20,
        "firstKillsDifference": -2,
        "flashAssists": 0,
        "hsKills": 14,
        "killDeathsDifference": -4,
        "kills": 16,
        "player": {
          "id": 339,
          "name": "ANGE1",
        },
        "rating1": 0.85,
      },
    ],
  },
  "result": {
    "halfResults": [
      {
        "team1Rounds": 10,
        "team2Rounds": 5,
      },
      {
        "team1Rounds": 3,
        "team2Rounds": 11,
      },
    ],
    "team1TotalRounds": 13,
    "team2TotalRounds": 16,
  },
  "roundHistory": [],
  "team1": {
    "id": 4775,
    "name": "Western Wolves",
  },
  "team2": {
    "id": 4555,
    "name": "Virtus.pro",
  },
}
`;
